app/src/main/java/com/sunasterisk/a14day_challenge/ui/listExercise/ListExercisesActivity.kt

```diff
@@ -2,18 +2,101 @@ package com.sunasterisk.a14day_challenge.ui.listExercise
 
 import android.content.Context
 import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.sunasterisk.a14day_challenge.R
+import com.sunasterisk.a14day_challenge.adapter.ListExercisesAdapter
+import com.sunasterisk.a14day_challenge.data.UserRepository
+import com.sunasterisk.a14day_challenge.data.local.DataBaseHandler
+import com.sunasterisk.a14day_challenge.data.local.UserLocalDataSource
+import com.sunasterisk.a14day_challenge.data.local.dao.UserDAOImp
+import com.sunasterisk.a14day_challenge.data.model.CurrentDay
+import com.sunasterisk.a14day_challenge.data.model.DetailExercise
+import com.sunasterisk.a14day_challenge.ui.ExerciseActivity
+import kotlinx.android.synthetic.main.activity_list_exercises.*
 
-class ListExercisesActivity : AppCompatActivity() {
+class ListExercisesActivity : AppCompatActivity(), ListExercisesContract.View,
+    ListExercisesAdapter.OnExerciseListener {
+
+    private var sharedPreferences: SharedPreferences? = null
+    private var presenter: ListExercisesPresenter? = null
+    private var myDataSet: ArrayList<DetailExercise>? = null
```

Comment:
Đặt tên rõ ràng, không sử dụng ArrayList, sử dụng List hoặc MutableList
Có thể khởi tạo ban đầu với 1 list rỗng để đỡ việc check null
Ưu tiên dùng val hơn var 
```suggestion
    private valclearNames = mutableListOf<DetailExercise>()
```